src/core/actions/cityactions/LevelUp.java

```diff
@@ -50,7 +50,7 @@ public boolean execute(GameState gs) {
                 city.addProduction(TribesConfig.CITY_LEVEL_UP_WORKSHOP_PROD);
                 break;
             case EXPLORER:
-                gs.getBoard().launchExplorer(cityPos.x, cityPos.y, city.getTribeId(), gs.getRandomGenerator());
+                gs.getBoard().launchExplorer(cityPos.x, cityPos.y, city.getTribeId(), gs.getRandomGenerator(), gs.getRandomGenerator());
```

Comment:
Why are you adding another random parameter here? Can we just use the 4th argument to this method? It's the same object.